packages/client/src/rpc.ts

```diff
@@ -28,30 +12,23 @@ export enum EventType {
 export interface RpcService {
   request: <R>(method: string, ...params: any[]) => Promise<R>
   addEventListener: (type: EventType, listener: EventListener) => void
+  close: () => void
 }
 
-export default (platform: Platform): RpcService => {
+export default (token: string, host: string, port: number): RpcService => {
   interface PromiseInfo {
     resolve: (value?: any) => void
     reject: (error: any) => void
   }
-
   const requests = new Map<ReqId, PromiseInfo>()
   let lastId = 0
 
   function createWebsocket (): Promise<WebSocket> { // eslint-disable-line @typescript-eslint/promise-function-async
-    const host = platform.getMetadata(core.metadata.WSHost) ?? 'localhost'
-    const port = platform.getMetadata(core.metadata.WSPort) ?? 3000
-
-    const token = platform.getMetadata(core.metadata.Token)
-
-    if (token === undefined) {
-      platform.broadcastEvent(PlatformStatus, new Status(Severity.ERROR, PlatformStatusCodes.AUTHENTICATON_REQUIRED, 'Authentication is required'))
-      return Promise.reject(new Error('authentication required'))
-    }
     return new Promise<WebSocket>((resolve) => {
       // Let's sure token is valid one
+      const WebSocket = (typeof window !== 'undefined' ? window.WebSocket : require('ws'))
```

Comment:
I do not like this: I believe both node and browser use cases could be properly designed to avoid such 'hacks'

Reply:
Added a RpcSocket to hold a implementation specific for different platforms.

Reply:
This is same: what is `RpcSocket` type? Can we create node WebSocket in node client and pass browser socket in plugin?

Reply:
they have incompatible interfaces, I've only could do so with using `any`.

Reply:
> they have incompatible interfaces

So current approach sucks more then :)

Reply:
I've found a variant I've like after some findings, it is actually possible to use package.json, "browser" field to specify a different source file for module, so for web we have variant with WebSocket under hood with types checked and for other we have 'ws' WebSocket version with all type checked calls. No hacks, but it uses intermediate RpcClient interface to unify some parts of code.

Reply:
Why do not do what you will do in statically typed language? You have to create `ws` socket from `client` package and `browser` from `platfrom-core`. So you do not need `browser` field.

Reply:
Get rid of browser field, and moved creation into platform-core and leave a node version inside index.ts of client.